### DISCUSSION TITLE
Build Script

### DISCUSSION THREAD

**Discussion by EdNekebno** [EdNekebno/DGTCentaurMods#72] [General]
Now that's an impressive build script...

I thought I'd test it. I thought it might be helpful to comment on what I did.

I copied the build folder to another folder and ran it. It handled everything great.

I got the buster raspbian lite and put it on my card and copied the .deb to /boot (coz I'm lazy and it was easiest and it fits :)
I checked that in my new RPi 3A+ and logged on and configured wifi. Then I think:

apt update
dpkg-i longfilename.deb
apt-get -f install

and it did many many things :) Then I rebooted.

I couldn't get the web interface. That may have been because I forgot the :5000 . But when I tried to run stuff, including app.py directly, it came up with an error about not being able to find the DGTCentaurMods module.

I'm lazy so I typed:

cd /usr/lib/python3/dist-packages
ln -s /home/pi/DGTCentaurMods DGTCentaurMods

reboot

and it ran just fine.

@wormstein - Is it worth me giving my DGT Centaur surgery for the millionth time ( :) ) and opening it up and putting this SD card in and seeing what works/doesn't, or am I better off waiting until you've done stockfish and whatever else you're planning? It looks excitingly close!

**Comment by wormstein** [comment]
I'll get this in order :)

You can install it right away with `sudo apt install ./<long_name>.deb`. This should run all the way to finish.

Second, I think the problem with not finding DGTCentaurMod is because PYTHONPATH is not set. Will have a look. It should be set up in the service file pointing to the setup location... Is anythng showing up in systemctl status?

I am about to finish all the stuff for stockfinsh. I have finished the option to build it in the deb if wanted. I just need to complete the second part, at installation. That will check if centaur is there and if not will create the path and push stockfish binary. If it's there, it will backup SF and replace it with our version.
What about the other files? rc.local, pin.conf, etc? Are they good? I might have to write down an uninstall procedure.

Just wait a little bit to open the board, at least till I am done with stockfish.

**Comment by EdNekebno** [comment]
It's amazing work.

I looked at the configuration files and they seem good. However, because I'm a bit clueless and it don't really know what I'm doing, I can't really tell until I stick the card in and I'm sure my board doesn't want opening up too much more :) PYTHONPATH was definitely in the service file, it  just didn't seem to work...but don't quote me on it!

One thing I realised is that I think some of the scripts use things like the chess font from the centaur folder. Then if centaur stuff isn't there it would break those. So I will look into if we can do our own versions of those assets or if the assets were freely available in the first place. Our own versions we can just put in the resources folder of course

**Comment by wormstein** [comment]
For me, last time it failed to start was the nasty epaper.jpg file :))
I'll have a look too.

I'm sure there is some font packages we could use to print to eink display. See what waveshare is reommending. Not sure what to say about other cliparts... Good catch though.

**Reply by EdNekebno** [reply]
epaper.jpg should be fixed.

It's not so much a font as a sprite sheet (I wasn't clear). There's a sprite sheet to draw the chess pieces every time it draws a board. I just referenced the centaur one. I'll build our own, shouldn't be too difficult.

**Reply by EdNekebno** [reply]
I fired up the sprite editor to do them copyright free. Queen is kinda hard to do small. But overall I think this will do. I'm not sure if I like darker black for the black squares or grey by doing every other pixel like the centaur had it.

![chesssprites](https://user-images.githubusercontent.com/82707095/139118077-d8f5d9d2-b736-4ed9-b31f-a225efae80b5.png)
..

**Reply by wormstein** [reply]
Looks pretty.

**Comment by wormstein** [comment]
After doing tens of builds, I have pushed [here](https://github.com/wormstein/DGTCentaur/tree/build) an updated build script that should add stockfish. Also I updated the installation process to check for centaur folder and do what's needed.

I now need to try it as a clean install but not sure if I have the enegy :))
One last thing I need to do is the uninstall script because weird things happen if you remove and reinstall the paclage withour staging everything back.

**Reply by EdNekebno** [reply]
Uninstall.....that sounds like something nobody is ever gonna want to do :)
I think the rest of this evening is time for Netflix watching, not coding.

**Comment by wormstein** [comment]
We'll need to delay this :(
I am now in the ER and might end up on the hospital for a week or so for a surgery...

**Reply by EdNekebno** [reply]
Poor you. :( Hope everything goes okay.

**Reply by wormstein** [reply]
Thanks. It's confirmed... See you in about a week. I'll still check in here from time to time.

Cheers. Have a break :)))

**Reply by EdNekebno** [reply]
Concentrate on healing. 

In the meantime I'll work on it but I won't push changes until we're ready for a build.

What I think is a plan is, when you are better (and only then! I'm guessing you're probably reading this coz you're bored :) ), then we work on the build. At the point we are ready to build it would be awesome for me to then update any changes I have, and for us to build the .deb and image on the same day and release. At that point I'll set up a dev branch so that we can consider the master the stable version.

**Reply by wormstein** [reply]
Hello everyone,
I had a horrific weekend, but i am doing better now.

Just wanted to let you know you could merge my build branch (create yourself a pull request ) to get latest changes.

Last change was related to catch the way debian's package manager is calling the maintenance scripts. See debian wiki for this for details. We need this to handle updates the best we can.

All the best!

**Reply by EdNekebno** [reply]
I see you're as bad a patient as I am! Thanks. But things can wait until you are actually recovered! You understand that stuff way better than me and I'd just hack it into some sort of monster frankenstein code. :)

**Reply by wormstein** [reply]
Hehe. Thanks man. It was a huge procedure indeed so I say about 10 days to recover... 

I did a pull request just in case...
Go ahead with branching to see how it works is a good idea. On release we will use tags to freeze a state of the  project.

**Reply by wormstein** [reply]
Hey, very good news.
Today I was discharged.

I a couple of days I am planning on building an alpha release to have something for end-user.

We will then need to write clear  install instructions. For now, in  case they want to reinstall, they will start from reimage of the RPiW. Once package maintenance scripts are done, they will handle all situations to avoid wiping the RPiW

**Reply by EdNekebno** [reply]
Awesome...take it slow though - best to be fully recovered.

If you let me know just before you are ready to build the alpha release then I will push some changes for you to include.
